param/ipython.py

```diff
@@ -124,6 +126,21 @@ def param_docstrings(self, info, max_col_len=100, only_changed=False):
         return "\n".join(contents)
 
 
+    def _ordered_params(self, parameters):
```

Comment:
Maybe call it `sort_by_precedence`, because (at least for py3) compared to the dictionary that's passed in, the dict stays the same except for a global ordering by precedence. I don't think it should be private; other libraries (e.g. Panel) should be able to use this, so it shouldn't start with an underscore.  A docstring could be "Sort the provided dictionary of parameters by their precedence value. In Python 3, preserves the original ordering for parameters with the same precedence; for Python 2 sorts them lexicographically by name.".

Reply:
Otherwise, looks good!